src/core/types/appearance_id.rs

```diff
@@ -124,6 +92,11 @@ mod tests {
         assert_ne!(SUT::sample(), SUT::sample_other());
     }
 
+    #[test]
+    fn len_of_all_is_max_plus_one() {
+        assert_eq!(SUT::all().len(), SUT::MAX as usize + 1);
```

Comment:
MAX is always _inclusive_ the max value. So with zero being a valid number, the number of elements is MAX+1. Feels natural for me

Reply:
Imagine if we only supported one gradient, we would have `MAX = 0`. We are also doing `MAX + 1` all over this file, think it would be easier to read and understand to consider MAX as the natural number and not the array index.

Reply:
We can call it COUNT instead if max. And validate value is ` < COUNT`

Reply:
Maybe we should ask for the rest of the team what their opinions are, but I still don't see any benefit in refering to numbers array-based rather than naturally.